middle_end/flambda2/flambda_middle_end.ml

```diff
@@ -80,47 +80,55 @@ let middle_end0 ppf ~prefixname ~backend ~filename ~module_ident
       ~module_block_size_in_words ~module_initializer =
   Misc.Color.setup (Flambda_features.colour ());
   Profile.record_call "flambda.0" (fun () ->
-    let flambda =
+    let flambda, code =
       Profile.record_call "lambda_to_flambda" (fun () ->
-        Lambda_to_flambda.lambda_to_flambda ~backend ~module_ident
-          ~module_block_size_in_words module_initializer)
+          Lambda_to_flambda.lambda_to_flambda ~backend ~module_ident
+            ~module_block_size_in_words module_initializer)
     in
     print_rawflambda ppf flambda;
     check_invariants flambda;
-    let flambda =
-      if Flambda_features.Debug.permute_every_name ()
-      then Flambda_unit.permute_everything flambda
-      else flambda
-    in
-    let round = 0 in
-    let new_flambda =
-      Profile.record_call ~accumulate:true "simplify"
-        (fun () -> Simplify.run ~backend ~round flambda)
-    in
-    if Flambda_features.inlining_report () then begin
-      let output_prefix = Printf.sprintf "%s.%d" prefixname round in
-      Inlining_report.output_then_forget_decisions ~output_prefix
-    end;
-    print_flambda "simplify" ppf new_flambda.unit;
-    output_flexpect ~ml_filename:filename flambda new_flambda.unit;
-    new_flambda)
+    if Flambda_features.Expert.fallback_inlining_heuristic ()
+    then
+      { cmx = None;
```

Comment:
Just as in the other branch, you should output the inlining_report before returning.